packages/core/storage-js/src/packages/StorageFileApi.ts

```diff
@@ -600,14 +600,21 @@ export default class StorageFileApi {
       if (this.shouldThrowOnError) {
         throw error
       }
-      if (isStorageError(error) && error instanceof StorageUnknownError) {
-        const originalError = error.originalError as unknown as { status: number }
+      if (isStorageError(error)) {
+        // Check for both StorageApiError (which has status directly) and StorageUnknownError (with originalError)
+        let status: number | undefined
+
+        if (error instanceof StorageApiError) {
+          status = error.status
+        } else if (error instanceof StorageUnknownError) {
+          const originalError = error.originalError as unknown as { status: number }
+          status = originalError?.status
+        }
 
-        if ([400, 404].includes(originalError?.status)) {
-          return { data: false, error }
+        if (status && [400, 404].includes(status)) {
+          return { data: false, error: null }
```

Comment:
I agree this is the most correct return, but this could be a breaking change for some people.

It seems very likely that at least some projects use code like this which would exhibit reveresed logic because of this change.

```typescript
const { data, error } = storage.exists('some/path.jpg)

if (error) {
  // potentially also checks if it's a NotFound error
  // do "not exists" stuff
}

// do "exists" stuff
```

I think we still have to return error instead of null to avoid a major breaking change.

Plus, the return type allows for it, and it's like saying "exists is false BECAUSE... (a 400 or 404 error)"

Reply:
@SumitKumar-17 what do you think, do you want to work on these changes?